src/heretic/mn_menu.c

```diff
@@ -612,6 +612,31 @@ static void InitFonts(void)
     FontBBaseLump = W_GetNumForName(DEH_String("FONTB_S")) + 1;
 }
 
+// [crispy] Check if printable character is existing in FONTA/FONTB sets.
+enum {
+    big_font, small_font
+} fonttype_t;
+
+static const char MN_CheckValidChar (char ascii_index, int have_cursor)
+{
+    if ((ascii_index >= 91 + have_cursor && ascii_index <= 96) || ascii_index >= 123)
+    {
+        // Replace "\]^_`" and "{|}~" with spaces,
+        // allow [ (91, cursor) only in small fonts.
+        return 32;
+    }
+    else if (ascii_index >= 97 && ascii_index <= 122)
```

Comment:
You could use char literals instead of magic numbers, i.e. `'a'` instead of `97` and `'z'` instead of `122`.